### ISSUE TITLE
Troubleshooting

### ISSUE THREAD

**Issue by deevus** [lvht/phpcd.vim#137]
I'm trying to get this working and I'm just getting nothing. 

What are some things to check to see if phpcd is running? I was able to run the command:
```
php ~/.config/nvim/pack/minpac/start/phpcd.vim/php/main.php `pwd` msgpack vendor/autoload.php 0
```
And no errors were thrown. 

I tried `:call phpcd#Index()` and I get no output. 
Cheers

## Snipped output of `php -m`
```
[PHP Modules]
msgpack
pcntl
```
## Versions
```
msgpack: 1.4.2-3
php: 7.1.15
php-msgpack: 2.0.2
neovim: 0.2.2
```

**Comment by lvht** [comment]
please make sure you phpcd process is running

![image](https://user-images.githubusercontent.com/2030591/37446199-b87a2918-2856-11e8-9540-36415ca25efd.png)

**Comment by deevus** [comment]
I can't see it in `htop`

**Comment by lvht** [comment]
please read the README carefully. make sure you project has a autoload php file.

**Comment by deevus** [comment]
I'm using composer so autoload.php is in the default location (`vendor/autoload.php`)

**Comment by lvht** [comment]
please check the `~/.phpcd.log` content

**Comment by deevus** [comment]
Sorry it doesn't exist

**Comment by lvht** [comment]
It is wired. please drop the two lines 
https://github.com/lvht/phpcd.vim/blob/master/php/main.php#L2
https://github.com/lvht/phpcd.vim/blob/master/php/main.php#L24

And run phpcd manually.

**Comment by deevus** [comment]
I'm not seeing any errors in the console when I try to invoke phpcd inside neovim.

**Comment by deevus** [comment]
If I press enter in the console while phpcd is running I get this:

```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Lvht\MsgpackRpc\ForkServer::onMessage() must be of the type array, integer given, called in /home/deevus/.config/nvim/pack/minpac/start/phpcd.vim/vendor/lvht/msgpack-rpc/src/ForkServer.php on line 32 and defined in /home/deevus/.config/nvim/pack/minpac/start/phpcd.vim/vendor/lvht/msgpack-rpc/src/AbstractServer.php:39
Stack trace:
#0 /home/deevus/.config/nvim/pack/minpac/start/phpcd.vim/vendor/lvht/msgpack-rpc/src/ForkServer.php(32): Lvht\MsgpackRpc\ForkServer->onMessage(10)
#1 /home/deevus/.config/nvim/pack/minpac/start/phpcd.vim/php/main.php(41): Lvht\MsgpackRpc\ForkServer->loop()
#2 {main}
  thrown in /home/deevus/.config/nvim/pack/minpac/start/phpcd.vim/vendor/lvht/msgpack-rpc/src/AbstractServer.php on line 39
```

Not sure if that's helpful

**Comment by mikemasam** [comment]
am getting same error here

**Comment by citlacom** [comment]
Hi @lvht  - I'm experimenting similar issue using MacOS, PHP 7.2 and NeoVim, I have installed the https://github.com/msgpack/msgpack-php extension and my project uses composer so it has autoload.php. It looks like the autostart of the phpcd service is failing. As @deevus commented in my local environment the ~/.phpcd.log file don't exists. I'm trying to start it manually but need to figure out the parameters to pass to main.php. I will update here if I'm able to find what's failing, maybe is an issue when using MacOS?

**Comment by lvht** [comment]
@citlacom echo all the [`rpc#start` args](https://github.com/lvht/phpcd.vim/blob/master/autoload/phpcd.vim#L1110) to get the main.php start flags.

And you need comment [this line](https://github.com/lvht/phpcd.vim/blob/master/php/main.php#L2) to see errors.

**Comment by citlacom** [comment]
Hi @lvht - Thanks for your help, finally I found the issue, as I have two versions of PHP 7 installed in my local environment, one that is from MAMP https://www.mamp.info/en/ which I use for web development and other that installed with Homebrew to satisfy the dependency pcntl dependency which don't come built in the MAMP PHP version. As I have exported in my environment PATH the MAMP bin directory I misunderstood that the variable g:phpcd_php_cli_executable can be set at my NeoVim init.vim so even when I set it to my Homebrew PHP version '/usr/local/opt/php@7.2/bin/php' when the phpcd server was started was still using the MAMP PHP which don't satisfy the "pcntl" dependency.

After look your example https://github.com/lvht/php-co-demo configuration move the php cli executable override to .phpcd.vim and that worked 👍 

Thanks for your support and great work on this plugin!
